src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1366,14 +1384,25 @@ subroutine von_Mises_calving(domain, err)
              if ( .not. li_mask_is_floating_ice(cellMask(iCell)) ) then
                 calvingVelocity(iCell) = sqrt(xvelmean(iCell)**2.0_RKIND + yvelmean(iCell)**2.0_RKIND) * &
                                          vonMisesStress(iCell) / config_grounded_von_Mises_threshold_stress
-             elseif ( li_mask_is_floating_ice(cellMask(iCell)) ) then
-                calvingVelocity(iCell) = sqrt(xvelmean(iCell)**2.0_RKIND + yvelmean(iCell)**2.0_RKIND) * &
+             elseif ( li_mask_is_floating_ice(cellMask(iCell)) .and. config_floating_von_Mises_threshold_stress .ne. 0.0_RKIND) then
+                calvingVelocity(iCell) = sqrt(xvelmean(iCell)**2 + yvelmean(iCell)**2) * &
                                          vonMisesStress(iCell) / config_floating_von_Mises_threshold_stress
+             !if gloating VM threshold is zero, remove floating ice
+             elseif ( li_mask_is_floating_ice(cellMask(iCell)) .and. config_floating_von_Mises_threshold_stress .eq. 0.0_RKIND &
+                       .and. li_mask_is_dynamic_ice(cellMask(iCell)) ) then
+                thickness(iCell) = 0.0_RKIND
```

Comment:
The convention in the calving module has been to never remove thickness directly, but instead populate the calvingThickness field first and then apply that to thickness later.  Is that possible to do here?

Reply:
calvingThickness is now populated for dynamic floating cells after calling apply_ablation_velocity if config_floating_von_Mises_threshold_stress is set to 0.0